### ISSUE TITLE
Using array actuals as field default values in nested types doesn't compile

### ISSUE THREAD

**Issue by e-kayrakli** [chapel-lang/chapel#18137]
```chapel
proc foo(A: []) {
  record Cmp {
    var arr = A;
  }

  return new Cmp();
}

writeln(foo([1,2,3]));
```

Fails with

```
internal error: AST-PRI-IVE-0235 chpl version 1.25.0 pre-release (c89e0f3496)
```

or with `--devel`:

```
internal error: invalid attempt to get reference type [AST/primitive.cpp:235]
```

This compiles if the argument is an integer.

**Comment by e-kayrakli** [comment]
This was reported by @npadmana while trying to workaround [another bug](https://github.com/chapel-lang/chapel/issues/18138).

**Comment by mppf** [comment]
FWIW I generally am surprised that nested records/classes are working at all.

**Comment by bradcray** [comment]
Is @npadmana blocked by these bugs?  (i.e., should we help look for a workaround, or prioritize fixing the issue?)

**Comment by e-kayrakli** [comment]
Not as far as I know: the code that led to this issue will be merged via https://github.com/chapel-lang/chapel/pull/18149 and it works around the bug by defining the record outside the function.

**Comment by npadmana** [comment]
Nope, this came up within the context of GSOC, and as Engin said, we have a good workaround.
